temporal-sdk/src/main/java/io/temporal/internal/replay/ReplayWorkflowContextImpl.java

```diff
@@ -202,7 +206,8 @@ public Functions.Proc1<Exception> scheduleActivityTask(
       attributes.setActivityId(workflowStateMachines.randomUUID().toString());
     }
     Functions.Proc cancellationHandler =
-        workflowStateMachines.scheduleActivityTask(parameters, callback);
+        workflowStateMachines.scheduleActivityTask(
```

Comment:
@Spikhalskiy As per our previous discussion, I passed `canRequestEagerExecution` down to the activity state machine, but I'm not that comfortable with going this deep. I think the check on `canRequestEagerExecution()` should be moved here, no deeper. What do you say?